### PR TITLE
[user] Add picture struct create endpoint

### PR DESCRIPTION
**Pull request by jaylees14**
Add support for a `picture` struct in the `user` service, which involves a couple of design decisions:

- Adds a new endpoint `user/<id>/picture` with CRUD defined on that (C now, RUD coming later)
- Implicit foreign key constraint on the picture table, but never returned
- Whenever a byte array is used it is NOT returned in the response body
- Byte arrays are expected to be encoded into base64 whenever used in JSON

Have added unit tests for this endpoint, but for those that like manual testing too:

```bash
ACCESS1=$(curl -s -X POST $BASE/api/auth/register -d "{\"email\":\"jay$RANDOM@test.com\", \"password\":\"abcdefgh\"}" | jq -r .AccessToken)

USER1=$(curl -s -X POST $BASE/api/user -d '{"name": "Jay"}' -H "Authorization: Bearer $ACCESS1" | jq -r .ID)

CAT=$(base64 ~/cat.png)

echo "{ \"img\": \"$CAT\" }" | curl -X POST $BASE/api/user/$USER1/picture -H "Authorization: Bearer $ACCESS1" -d @-

```

### CODE REVIEW

**Review by Lewiky — APPROVED** [review]
Cool cool cool

File: user/dao/dao.go [hunk 1]
```diff
@@ -136,3 +155,22 @@ func (dao *DAO) DeleteUser(input DeleteUserInput) error {
 
 	return nil
 }
+
+// CreatePicture new picture in the datastore, returning the newly created picture
```
Comment (willjvsmith) [inline user/dao/dao.go:159]: Do we want the DAO to return the picture?

File: user/user.go [hunk 2]
```diff
@@ -358,3 +370,89 @@ func (env *env) deleteUserHandler(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
+	userID, err := util.ExtractIDFromRequest(mux.Vars(r))
+	if err != nil {
+		respondWithError(w, err.Error(), http.StatusBadRequest, metric.RequestCreatePicture)
+		return
+	}
+
+	// Only the auth that created the user can create a single picture
```
Comment (willjvsmith) [inline user/user.go:387]: Strange sentence, but I understand why 😄

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀 

Would have liked a `cat.png` for the manual test

File: user/dao/dao.go (hunk 1 above)
Comment (jaylees14) [reply user/dao/dao.go:159]: I just did it for consistency, but agree we probably don't need it. Can remove it - what do you think?

**Pull request merged**